homeassistant/loader.py

```diff
@@ -63,6 +63,76 @@ def manifest_from_legacy_module(domain: str, module: ModuleType) -> Dict:
     }
 
 
+def manifest_from_path(path: pathlib.Path) -> Optional[Dict]:
+    """Try to load a manifest from a component path.
+
+    Will return None if no manifest exist or if it is malformed
+    """
+    try:
+        manifest_path = path / "manifest.json"
+        with manifest_path.open() as data:
+            manifest = json.load(data)  # type: Dict
+            return manifest
+    except FileNotFoundError:
+        return None
+    except ValueError as err:
+        _LOGGER.error("Error parsing manifest.json file at %s: %s",
+                      manifest_path, err)
+        return None
+
+
+async def get_custom_flows(hass: 'HomeAssistant') -> List:
```

Comment:
I wonder if instead of this , we should add a method to just load all custom integrations and return in a list. Iterate over dir like now and use resolve_from_root(custom_components, name)

Reply:
Otherwise we're going to add that method too in the future probably

Reply:
Right. Do you want the list of Integrations cached instead of the list of domains with config_flow?

Reply:
An aside. Currently the list of config flows are expected to be a sorted list. so if we cache the integrations, we'd need to re-unique:ify and sort the list of config_flows each time (unless we cache that too)

Reply:
I moved cache. The list of flows is no that heavy to calculate on the fly